web_monitoring/cli.py

```diff
@@ -60,11 +464,185 @@ def _filter_unchanged_versions(versions):
             yield version
 
 
+def _list_ia_versions_for_urls(url_patterns, from_date, to_date,
+                               skip_repeats=True, version_filter=None,
+                               client=None, stop=None):
+    version_filter = version_filter or _is_page
+    skipped = 0
+
+    with client or ia.WaybackClient() as client:
+        for url in url_patterns:
+            if stop and stop.is_set():
+                break
+
+            ia_versions = client.list_versions(url,
+                                            from_date=from_date,
+                                            to_date=to_date,
+                                            skip_repeats=skip_repeats)
+            try:
+                for version in ia_versions:
+                    if stop and stop.is_set():
+                        break
+                    if version_filter(version):
+                        yield version
+                    else:
+                        skipped += 1
+                        logger.debug('Skipping URL "%s"', version.url)
+            except ia.BlockedByRobotsError as error:
+                logger.warn(str(error))
```

Comment:
Include the exception type.
```suggestion
                logger.warn(repr(error))
```

Reply:
In this case, I didn’t do that because this exception always has nice descriptive text: https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/98daf6360dea0bec4db4f8a4e3242e07740a490b/web_monitoring/internetarchive.py#L501-L503

Do you think we should still use `repr` here? Should I cut down the error message if so?

Reply:
Side note: I guess I should move most of that to the actual error class instead of here where we raise it.

Reply:
OK, doesn't seem like including the exception type in the warning would add much here then.

Reply:
Argh, went to move the text and remembered why I didn’t originally do that: we could probably end up encountering a `robots.txt` issue in other API besides CDX as well (although I’ve only seen it and know how it’s formatted in CDX).

Any thoughts on how to better design this for that? Should we let the constructor take some extra info like:

```py
raise BlockedByRobotsError(query['url'], f'In CDX search {final_query}')
```

Should we leave it to where the error is handled?

```py
try:
    raise ia.BlockedByRobotsError(query['url'])
except ia.BlockedByRobotsError as error:
    logger.warn(f'CDX search error: {error!r}')
```

Or something else? Leave it as-is?

Reply:
I see. I like the snippet above because it’s the job of the caller to explain what it was doing that caused the error. I don’t think exception’s constructor need be extended to carry arbitrary contextual information.

Reply:
OK, I took a crack at that in 91e48ce.